tests/ecdhp256-test.c

```diff
@@ -89,8 +93,134 @@ int main()
 	ok = ok && compare_and_print(64, point_compressed, compressed7);
 
 
+	printf("%s\n", "Wycheproof tests ECDH: ");
+		
+	for (int i = 0 ; i< sizeof(w_vectors)/sizeof(ecdhp256_w_i); i++)
+	{
+		printf("ECDH Wycheproof Test %d ", i );
+
+		uint8_t* decompressedPoint = (uint8_t*) malloc (sizeof (uint8_t) * 64);
+		bool flagDecompress = Hacl_P256_decompression_not_compressed_form(w_vectors[i].publicKey, decompressedPoint);
+
+		uint64_t success = Hacl_P256_ecp256dh_r(result, decompressedPoint, w_vectors[i].privateKey);
+
+		uint64_t fDU = 0;
+		if (flagDecompress = true)
+			fDU = success;
+		
+		if (w_vectors[i].flag != 1) 
+		{
+
+			ok = ok && ((success & fDU) == w_vectors[i].flag);
+			if (success == 0) {
+				printf("%s\n", "");
+				ok = ok && compare_and_print(32, result, w_vectors[i].sharedKey);
+				}
+		}
+		else
+			if (success == 0){
+				printf("%s\n", "");
+				ok = ok && compare_and_print(32, result, w_vectors[i].sharedKey);
+			}
+
+
+		if (!ok) 
+			{
+				printf("Test %d \n failed", i);
+				break;
+			}
+		else
+			printf("-- success \n");
+
+	}
+
+
+	printf("%s\n", "Wycheproof tests ECDSA: ");
+		
+	for (int i = 0 ; i< sizeof(w_ecdsa_vectors)/sizeof(ecdsap256_w_i) ; i++)
+	{
+		printf("ECDSA Wycheproof Test %d", i );
+		
+		uint32_t mLen = w_ecdsa_vectors[i].mLen;
+		
+		static uint8_t pk_ [64] = {0};
+		for (int j = 0; j < 32; j++)
+		{
+			pk_[j] = w_ecdsa_vectors[i].publicX[j];
+			pk_[j+32] = w_ecdsa_vectors[i].publicY[j];
+		}
+
+
+		bool resultVerification2 = Hacl_P256_ecdsa_verif_p256_sha2(mLen, w_ecdsa_vectors[i].message, pk_, w_ecdsa_vectors[i].r, w_ecdsa_vectors[i].s);	
+		
+		if ((w_ecdsa_vectors[i].flag == 18446744073709551615U) && (resultVerification2 == true))
```

Comment:
please don't use magic constants and #define them with a meaningful name

Reply:
All done :)